### PR TITLE
Maintenance/fair software

### PR DESCRIPTION
**Pull request by mcw92**
# Description

This PR implements the [FAIR software action](https://github.com/fair-software/howfairis-github-action) which assesses the compliance of the ARTIST codebase with [fair-software.eu](https://fair-software.eu/). The action runs upon pushes to the main branch.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings

Current status
```
Checking compliance with fair-software.eu...
url: https://github.com/ARTIST-Association/ARTIST
(1/5) repository
      ✓ has_open_repository
(2/5) license
      ✓ has_license
(3/5) registry
      × has_ascl_badge
      × has_bintray_badge
      × has_conda_badge
      × has_cran_badge
      × has_crates_badge
      × has_maven_badge
      × has_npm_badge
      × has_pypi_badge
      × has_rsd_badge
      × is_on_github_marketplace
(4/5) citation
      × has_citation_file
      × has_citationcff_file
      × has_codemeta_file
      × has_zenodo_badge
      × has_zenodo_metadata_file
(5/5) checklist
      × has_core_infrastructures_badge

Calculated compliance: ● ● ○ ○ ○

```

To improve the fairness of ARTIST, we should:
- Add a CITATION.cff file. This can easily be generated with https://citation-file-format.github.io/cff-initializer-javascript/#/
- Add a PyPI badge once ARTIST is on PyPI.
- Add a Zenodo badge / doi once ARTIST is on Zenodo.

### CODE REVIEW

**Review by kalebphipps — COMMENTED** [review]
Looks good to me!

Once we have the citation.cff the I am more than happy to merge!

**Comment by Filos1992** [comment]
Added an updated citation.cff file :) can be closed, except we want to wait for pypi?

**Comment by mcw92** [comment]
@Filos1992 Thanks for adding a citation file. However, I just found that you added a `CITATION.cff` for PAINT to ARTIST. While we can certainly use this file in the PAINT repo, we need another one for ARTIST itself. Please fix that (as well as the typos in the uploaded file before migrating it to PAINT).

**Comment by mcw92** [comment]
@Filos1992 There were still things wrong in the CITATION.cff file... I fixed them myself now and assigned @kalebphipps for review.

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    email: robert.pitz-paal@dlr.de
+    affiliation: German Aerospace Center (DLR)
+  - given-names: Moritz
+    family-names: Leibauer
+    email: moritz.leibauer@synhelion.com
+    affiliation: Synhelion Germany
+  - given-names: Markus
+    family-names: Götz
+    email: markus.goetz@kit.edu
+    affiliation: Karlsruhe Institute of Technology (DLR)
```
Comment (kalebphipps) [inline CITATION.cff:57]:
```suggestion
    affiliation: Karlsruhe Institute of Technology (KIT)
```
The abbreviation of the affiliation should be KIT

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    orcid: 'https://orcid.org/0000-0002-2233-1041'
+repository-code: 'https://github.com/ARTIST-Association/ARTIST'
+url: 'https://artist.readthedocs.io/en/latest/?badge=latest'
+abstract: >-
+  The ARTIST package provides an implementation of a
+  differentiable ray tracer using the PyTorch
+  machine-learning framework in Python. Leveraging automatic
+  differentiation and GPU computation, it facilitates the
+  optimization of heliostats, towers, and camera parameters
```
Comment (kalebphipps) [inline CITATION.cff:66]: maybe here "solar towers" instead of just towers?

**Review by kalebphipps — APPROVED** [review]
One typo for the affiliation should be changed but apart from that everything looks good - I am happy to approve the PR!

File: CITATION.cff (hunk 2 above)
Comment (mcw92) [reply CITATION.cff]: Fixed :).

**Pull request merged**